src/deadline/job_attachments/models.py

```diff
@@ -469,6 +474,32 @@ class PathMappingRule:
     def get_hashed_source_path(self, hash_alg: HashAlgorithm) -> str:
         return hash_data(self.source_path.encode("utf-8"), hash_alg)
```

Comment:
This needs to include the location name if it's available too, not just the root path. It needs to produce the same value as https://github.com/aws-deadline/deadline-cloud/blob/mainline/src/deadline/job_attachments/asset_sync.py#L536